### PR TITLE
Changelog updates

### PR DESCRIPTION
**Pull request by mikeday0**
Crispy friends: here's my humble attempt to catalog all the changes over the last year. You folks have been busy! Please review for wording and correctness. Thanks!

@JNechaevsky @rfomin @ceski-1 @SoDOOManiac @tomas7770 @SilverMiner @kitchen-ace

### CODE REVIEW

**Comment by kitchen-ace** [comment]
Nice work, bookkeeping is never anyone's favourite job. 

Can't speak to everything but anything with my name on it looks good, and I didn't spot any errors elsewhere at a glance.

I'll see if I can fix [this](https://github.com/fabiangreffrath/crispy-doom/commit/cbd56ecdba31ad08dc7269fcda792b79d8dcc034#commitcomment-138507807) today.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,135 +1,79 @@
...
-  * SysEx messages.
-  * Proper device reset (winmm_reset_type and winmm_reset_delay config options).
-  * Loop points (Final Fantasy and RPG Maker).
-  * Full support of EMIDI.
-* All music formats now work when the OPL backend is selected (@rfomin).
-* Implement demo footer for Doom and Heretic (@rfomin).
+**General Fixes and Improvements**
+* Don't apply framecap when using `-timedemo` (@JNechaevsky).
+* Make demo footer compatible with PrBoom+/DSDA-Doom demo autoplay (@rfomin).
+* Use "Crispier" default setting for Windows native midi (@ceski-1).
```
Comment (ceski-1) [inline CHANGELOG.md:4]: This can be removed, it's not important.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,135 +1,79 @@
...
-* Add weapon bobbing interpolation from Woof! (@rrPKrr).
-* Restore colored blood options from previous Crispy Doom versions (@rrPKrr).
-* Add known hashes for Sigil music tracks (@SirYodaJedi).
-* Colored blood setting to change in both directions (@SoDOOManiac).
-* Minor HUD fixes (@SoDOOManiac).
+* Improve display of secret sector coloring in automap when custom PLAYPALs are
+  used (@SoDOOManiac).
+* Use PWAD-provided color translation tables for colored blood.
+* "Status Bar" level stats are now left-aligned with HUD (@SoDOOManiac).
+* Fix visual glitch on door in Doom 2 MAP19 when running uncapped (@rfomin).
```
Comment (rfomin) [inline CHANGELOG.md:22]: @JNechaevsky helped here as well.

Thanks!

**Comment by JNechaevsky** [comment]
Composing changelog is always most unexciting thing to do, thank you very much @mikeday0! I made few tiny corrections to make things a little bit fair. 🙂

One thing that disturbs be: I have `Program:Win32/Wacapew.C!ml` false alarm about compiled `crispy-doom.exe` in Microsoft Defender check on VirusTotal. It's slightly different from Woof, but have same roots of the problem. But! If `crispy-doom.exe` is processed via `strip`, then no false alarm is happening. Maybe this will helps somehow?

**Comment by rfomin** [comment]
> If `crispy-doom.exe` is processed via `strip`, then no false alarm is happening. Maybe this will helps somehow?

We should run `strip` during packaging:
https://github.com/fabiangreffrath/crispy-doom/blob/d80f4a110b01c502276a3db0e1dac47af692b32c/pkg/win32/GNUmakefile#L53

**Review by fabiangreffrath — APPROVED** [review]
Looks good, thank you!

**Comment by JNechaevsky** [comment]
Do we have an established release date yet? I have a small set of improvements for Hexen, collected while working on Inter, and definitely would like to contribute them to Crispy. Nothing too big, but I would like to ask for one or two extra evenings to don't do anything in rush.

**Comment by fabiangreffrath** [comment]
No, not in a rush at all. I'd like to get the Choco release through the door first, so we'd have a new clean base to biuld upon.

**Comment by mikeday0** [comment]
Shall I leave this open for additional items then?

**Comment by fabiangreffrath** [comment]
Now that Chocolate Doom 3.1.0 has been released (🥳) I'd kindly like to ask all Crispy contributors to get their changelogs updated, so we don't have to wait too long to follow suit.

**Comment by SoDOOManiac** [comment]
@fabiangreffrath, I have just a tiny pull request:
https://github.com/fabiangreffrath/crispy-doom/pull/1189

**Comment by SoDOOManiac** [comment]
And yeah, another one, tiny as well
https://github.com/fabiangreffrath/crispy-doom/pull/1202

**Comment by JNechaevsky** [comment]
Wrote changes from my side. Not all of them, just important, I have dropped some technical ones to keep things clean. But please, feel free to correct me if needed, I was bad at composing changelogs 7 years ago at starting my "source port cariere", and bad with it even now. 🙃

Probably worth to make a note about TrueColor support for all games in the first lines of release post, but I'd like to leave this honor to @fabiangreffrath. 

_P.S. Barely I can find a words to describe amount of TrueColor abilites. It's not just a `16,777,216` colors mode. It's a **God**send._

**Comment by JNechaevsky** [comment]
Uh-oh. Woke up with a though that Hexen needs small update of drawing translucent functions as well, just like it was done for [Heretic](https://github.com/fabiangreffrath/crispy-doom/commit/dcb8585fe24a34f637bd7d535ccc8919e17dae31#diff-0256e7fa9285dbf859042680577a054534ecc5b6d8553076d1b45d843a6e60f8R181). PSP translucency is not an often guest there, but it's used for Cleric's invulnerability effect (missing one pixel line at the bottom):

<details>

image

</details>

There are two drawing functions, normal and alt., I'll take care about them shortly.

**Comment by fabiangreffrath** [comment]
All done here? Then I'd like to merge this tomorrow and kick off another release based on the brand new Chocolate Doom 3.1.0!

**Comment by mikeday0** [comment]
> All done here? Then I'd like to merge this tomorrow and kick off another release based on the brand new Chocolate Doom 3.1.0!

Nothing more from me!

**Comment by fabiangreffrath** [comment]
Alright, I expect to kick it off on Friday.

**Comment by JNechaevsky** [comment]
One more thing, please. @kitchen-ace [suggested](https://www.doomworld.com/forum/topic/112333-this-is-woof-1450-apr-30-2024/?do=findComment&comment=2827883) to have a sound fix for huge levels, and probably it will be good addition for `crispy_soundfix`. Easily to implement, but first approval of @fabiangreffrath is needed. Suggested aproach is something like this:

<details>

```diff
+++ D:/JN/GitHub/crispy-doom/src/doom/p_setup.c	Wed Nov  8 10:21:22 2023
@@ -887,17 +887,8 @@
 	}
 
 	// set the degenmobj_t to the middle of the bounding box
+	if (!crispy->soundfix)
+	{
 	sector->soundorg.x = (bbox[BOXRIGHT]+bbox[BOXLEFT])/2;
 	sector->soundorg.y = (bbox[BOXTOP]+bbox[BOXBOTTOM])/2;
+	}
+	else
+	{
+	// [crispy] Andrey Budko: fix sound origin for large levels
+	sector->soundorg.x = bbox[BOXRIGHT]/2+bbox[BOXLEFT]/2;
+	sector->soundorg.y = bbox[BOXTOP]/2+bbox[BOXBOTTOM]/2;
+	}
 		
 	// adjust bounding box to map blocks
 	block = (bbox[BOXTOP]-bmaporgy+MAXRADIUS)>>MAPBLOCKSHIFT;

```

</details>

Also, I think there is a way to support custom palette tinting effects in TrueColor. In theory, they shouldn't be `color_panes`, but colormap arrays, same to `pal_color`. This should do the trick, but:

- a) This ~~may~~ will lead to possibly way too big diff comparing to current implementation and more `ifndef`'s.
- b) For god's sake, not now, no need to delay release because small abstract thought.

**Comment by fabiangreffrath** [comment]
> Suggested aproach is something like this:

Yes, please. This will fit the current release quite well. 

> Also, I think there is a way to support custom palette tinting effects in TrueColor.

Maybe, but later. We have really reached the point for a new release.

**Comment by fabiangreffrath** [comment]
> Wait, what if `I_SetPalette` will set/swap pointers between different `colormaps[]` arrays?.

You'd still have to redraw everything that's already on screen, i.e. status bar, bezel, crosshair, ...

**Comment by JNechaevsky** [comment]
Ah, yes, that's a `pal_color[]` array implemented for a good reason. And even if it will be precalculated/tablified as well, _probably_ a full screen refresh will still be needed, as some parts of screen are using "buffered" drawing, like screen bezel and wide screen status bar sides. 

If done carefully, having an extra 13 arrays for `colormaps[]` + 13 arrays for `pal_color[]` (but, ahem, it's +27/+27 in Hexen palettes!) will probably help to get few extra fps on palette changing and provide some extra compatibility, but... SDL color operations for color panes is not that expensive, and the idea itself doesn't sounds too reasonable. Maybe one day later, if it will not require too much code changes and/or destruction, perhaps it will deserve a chance to suggest and live. But as decided, not now of course.

My another idea was [to use](https://github.com/JNechaevsky/international-doom/commit/04b042744da60c90fe0ded96c1518f36b28056ce) `malloc`/`free` for wiping effects, since `(d)`/`pixel_t` is consuming more memory in TrueColor, and such memory becomes a hostage in purgeable zone memory, while it can be freed immediately after wipe effect. But that's it, enough for now. 🤐

**Pull request merged**